source/slang/slang-mangle.cpp

```diff
@@ -547,6 +547,18 @@ namespace Slang
 
             for(auto paramDeclRef : parameters)
             {
+                // "out"/"inout" and "in" modifier makes big difference in the spirv code generation,
+                // because "out"/"inout" parameter will be passed by pointer. Therefore, we need to
+                // distinguish them in the mangled name to avoid name collision.
+                if(paramDeclRef.getDecl()->hasModifier<OutModifier>() ||
```

Comment:
We need a different prefix for each different direction. There should already be a getParameterDirection function you can use. in, inout and out and ref, const ref should all get different mangled names.

Reply:
done.